protocol/sip/helper.go

```diff
@@ -0,0 +1,509 @@
+// Package sip is a very basic (and incomplete) implementation of SIP messaging protocol.
+//
+// Based on the sipgo library, we are only adding some helpers useful in the context of exploitation.
+// For example, we prefer to avoid the complexity of concepts like transactions or dialogs.
+//
+// References:
+// - https://github.com/emiago/sipgo
+// - https://datatracker.ietf.org/doc/html/rfc3261
+// - https://datatracker.ietf.org/doc/html/rfc3311
+// - https://datatracker.ietf.org/doc/html/rfc3581
+// - https://datatracker.ietf.org/doc/html/rfc3265
+// - https://datatracker.ietf.org/doc/html/rfc7118
+package sip
+
+import (
+	"bufio"
+	_ "embed"
+	"fmt"
+	"io"
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/emiago/sipgo/sip"
+	"github.com/google/uuid"
+	"github.com/vulncheck-oss/go-exploit/output"
+	"github.com/vulncheck-oss/go-exploit/protocol"
+)
+
+const (
+	// SIP over UDP message size should be lower than 1300 bytes.
+	// https://datatracker.ietf.org/doc/html/rfc3261#section-18.1.1
+	UDPMessageLength         = 1300
+	errMsgRequiredParam      = "Required parameter: %s"
+	DefaultMaxForwards       = 70
+	DefaultCSeq              = 1
+	DefaultInviteContentType = "application/sdp"
+	DefaultInviteBodyContent = `v=0
+o=caller 123456 654321 IN IP4 192.168.1.100
```

Comment:
Should any of these IP addresses or content be tied to any variables, or does it not matter much? I could add a `random.RandIP` for some degree of randomness.